pandas/tests/groupby/test_rank.py

```diff
@@ -290,3 +290,17 @@ def test_rank_empty_group():
     result = df.groupby(column).rank(pct=True)
     expected = DataFrame({"B": [0.5, np.nan, 1.0]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_rank_zero_div():
+    # GH 23666
+    df = pd.DataFrame({
+        "A": [1, 1, 1, 2, 2, 2],
```

Comment:
Can you make this test case any smaller? As minimal of an example to produce the error as possible would be clearest